### PR TITLE
Ensure AssetCodes are encoded as expected by Stellar

### PR DESCRIPTION
**Pull request by juanhurtado10**
This PR complements the #51

### CODE REVIEW

File: lib/xdr/assets/asset_code12.ex [hunk 1]
```diff
@@ -4,46 +4,76 @@ defmodule Stellar.XDR.AssetCode12 do
...
-  def encode_xdr!(%__MODULE__{code: code}) do
+  def encode_xdr!(%__MODULE__{code: code, length: length}) when length in @length_range do
     code
-    |> XDR.VariableOpaque.new(@max_size)
-    |> XDR.VariableOpaque.encode_xdr!()
+    |> XDR.FixedOpaque.new(length)
+    |> XDR.FixedOpaque.encode_xdr!()
   end
 
+  def encode_xdr!(_asset_code), do: raise(XDR.FixedOpaqueError, :invalid_length)
```
Comment (luishurtado) [inline lib/xdr/assets/asset_code12.ex:32]: Not sure if we should expose the `XDR.FixedOpaqueError`, which is an internal error returned by the XDR library. Something like `Stellar.XDR.AssetCode12Error` could be more appropriate. What do you think?

Comment (juanhurtado10) [reply lib/xdr/assets/asset_code12.ex:32]: @luishurtado definitely, agree with you!

**Review by luishurtado — COMMENTED** [review]
Let's discuss how to name errors, see comment.

File: lib/xdr/assets/asset_code12.ex (hunk 1 above)
Comment (juanhurtado10) [reply lib/xdr/assets/asset_code12.ex]: @luishurtado The Exceptions modules were implemented. Could you please check again? thanks!

**Review by luishurtado — APPROVED** [review]

**Pull request merged**